cf/read_write/netcdf/netcdfread.py

```diff
@@ -335,51 +348,70 @@ def _create_Data(
             chunks=chunks,
             **kwargs,
         )
-        self._cache_data_elements(data, ncvar)
 
         return data
 
-    def _customize_read_vars(self):
-        """Customize the read parameters.
+    def _customise_read_vars(self):
+        """Customise the read parameters.
+
+        Take the opportunity to apply CFA updates to
+        `read_vars['variable_dimensions']` and
+        `read_vars['do_not_create_field']`.
 
         .. versionadded:: 3.0.0
 
         """
-        super()._customize_read_vars()
-
+        super()._customise_read_vars()
         g = self.read_vars
 
+        if not g["cfa"]:
+            return
+
+        g["cfa_aggregated_data"] = {}
+        g["cfa_aggregation_instructions"] = {}
+        g["cfa_file_substitutions"] = {}
+
         # ------------------------------------------------------------
-        # Find out if this is a CFA file
+        # Still here? Then this is a CFA-netCDF file
         # ------------------------------------------------------------
-        g["cfa"] = "CFA" in g["global_attributes"].get("Conventions", ())
+        if g["CFA_version"] < Version("0.6.2"):
+            raise ValueError(
+                f"Can't read file {g['filename']} that uses obselete "
+                f"CFA conventions version CFA-{g['CFA_version']}. "
+                "(Note that version 3.13.1 can be used to read and "
```

Comment:
```suggestion
                "(Note that cf version 3.13.1 can be used to read and "
```
just to be super clear since we're mentioning CFA versions too?